src/components/Icon/BaseIcon.tsx

```diff
@@ -27,44 +29,23 @@ export interface IconProps extends BaseComponentProps {
   size?: IconSize;
 }
 
-export interface IconStyles {
-  height?: string;
-  width?: string;
-  color?: string;
-}
-
 /**
  * An `Icon` renders an SVG icon.
  */
-export default class BaseIcon extends React.Component<IconProps> {
-  protected getClassName() {
-    const { block, className } = this.props;
-
-    const classes = ['y-icon'];
-    if (block) {
-      classes.push('y-icon__isBlock');
-    }
-    if (className) {
-      classes.push(className);
-    }
-
-    return classes.join(' ');
+export default abstract class BaseIcon extends React.Component<IconProps> {
+  public render() {
+    return (
+      <TextSizeContext.Consumer>
+        {textSizeContext => {
+          const textSize = textSizeContext.textSize;
+          const { block, className, size, color } = this.props;
+          const classNames = getClassNames({ block, size, color, textSize });
+
+          return this.renderIcon(join(['y-icon', className, classNames.root]));
+        }}
+      </TextSizeContext.Consumer>
```

Comment:
I chose to bring the `render` function to the BaseIcon to consume `TextSizeContext` rather than putting it in the template, because I thought it would be a bit easier to change/maintain.  

The `render` function that was defined in the template has been renamed to `renderIcon`.